src/dmd/parse.d

```diff
@@ -8919,14 +8919,18 @@ final class Parser(AST) : Lexer
         endtok = token.value == TOK.leftBracket ? TOK.rightBracket : TOK.rightParentheses;
 
         nextToken();
+
         while (token.value != endtok && token.value != TOK.endOfFile)
         {
             auto arg = parseAssignExp();
             arguments.push(arg);
-            if (token.value == endtok)
+            if (token.value == endtok || token.value != TOK.comma)
```

Comment:
As `endtok` is never a comma, the first part of the condition is unnecessary now. The braces below can also be omitted.
Other than that, LGTM.

Reply:
is it worth checking for endTok before the loop and switching to while(true)?

Reply:
A trailing comma is optional, so the check in the while condition should be kept.